File: src/leetcode/interview/easy/122. Best Time to Buy and Sell Stock II/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int size = prices.size();
        if(size == 0) return 0;
        
        int result = 0;
        for(int i = 0; i < size - 1; i++)
        {
            int r = prices[i + 1] - prices[i];
            if(r > 0)
            {
                result += r;     
            }
        }
        
        return result;
    }
};

File: src/leetcode/interview/easy/122. Best Time to Buy and Sell Stock II/solution.go
func maxProfit(prices []int) int {
    if len(prices) == 0 {
        return 0
    }
    
    ans := 0
    
    for i := 0; i < len(prices) - 1; i++ {
        result := prices[i + 1] - prices[i]
        
        if result > 0 {
            ans += result
        }
    }
    
    return ans
}
